common/config.go

```diff
@@ -0,0 +1,240 @@
+	WriteTimeout      uint32 `yaml:"write-timeout"`
+	NodeLimit         uint32 `yaml:"node-limit"`
+}
+
+type DeviceControllerSt struct {
+	Kube KubeEdgeCntrllerConfig `yaml:"kube"`
+}
+
+type ControllerYaml struct {
+	Controller       CloudControllerSt  `yaml:"controller"`
+	CloudHub         CloudHubSt         `yaml:"cloudhub"`
+	DeviceController DeviceControllerSt `yaml:"devicecontroller"`
+}
+
+type ModulesSt struct {
+	Enabled []string `yaml:"enabled"`
+	//Enabled string `yaml:"enabled"`
+}
+
+type ModulesYaml struct {
+	Modules ModulesSt `yaml:"modules"`
+}
+
+type LoggingYaml struct {
+	LoggerLevel   string   `yaml:"loggerLevel,omitempty"`
+	EnableRsysLog bool     `yaml:"enableRsyslog,omitempty"`
+	LogFormatText bool     `yaml:"logFormatText,omitempty"`
+	Writers       []string `yaml:"writers,omitempty"`
+	LoggerFile    string   `yaml:"loggerFile,omitempty"`
+}
+
+func Write2File(path string, data interface{}) error {
+	d, err := yaml.Marshal(&data)
+	if err != nil {
+		return err
+	}
+
+	if err = ioutil.WriteFile(path, d, 0666); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteControllerYamlFile(path string) error {
+	controllerData := ControllerYaml{Controller: CloudControllerSt{Kube: KubeEdgeCntrllerConfig{Master: "http://localhost:8080", Namespace: "",
+		ContentType: "application/vnd.kubernetes.protobuf",
+		QPS:         5, Burst: 10, NodeUpdateFrequency: 10,
+		Kubeconfig: ""}},
+		CloudHub: CloudHubSt{IPAddress: "0.0.0.0", Port: 10000, CA: "/etc/kubeedge/ca/rootCA.crt", Cert: "/etc/kubeedge/certs/edge.crt",
+			Key: "/etc/kubeedge/certs/edge.key", KeepAliveInterval: 30, WriteTimeout: 30, NodeLimit: 10},
+		DeviceController: DeviceControllerSt{Kube: KubeEdgeCntrllerConfig{Master: "http://localhost:8080", Namespace: "", ContentType: "application/vnd.kubernetes.protobuf",
+			QPS: 5, Burst: 10, Kubeconfig: ""}},
+	}
+	if err := Write2File(path, controllerData); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteCloudModulesYamlFile(path string) error {
+	modulesData := ModulesYaml{Modules: ModulesSt{Enabled: []string{"devicecontroller", "controller", "cloudhub"}}}
+	if err := Write2File(path, modulesData); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteCloudLoggingYamlFile(path string) error {
+	loggingData := LoggingYaml{LoggerLevel: "INFO", EnableRsysLog: false, LogFormatText: true, Writers: []string{"file", "stdout"}, LoggerFile: "edgecontroller.log"}
+	if err := Write2File(path, loggingData); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteEdgeLoggingYamlFile(path string) error {
+	loggingData := LoggingYaml{LoggerLevel: "DEBUG", EnableRsysLog: false, LogFormatText: true, Writers: []string{"stdout"}}
+	if err := Write2File(path, loggingData); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteEdgeModulesYamlFile(path string) error {
+	modulesData := ModulesYaml{Modules: ModulesSt{Enabled: []string{"eventbus", "servicebus", "websocket", "metaManager", "edged", "twin", "dbTest"}}}
```

Comment:
Is dbTest required to load?

Reply:
Yes, it is not required and shall be removed